Лабораторная работа 1 Исправленная

MrGriv opened the pull request.

MrGriv committed:
Лабораторная работа 1

MrGriv committed:
Update main.py

frek17 commented on main.py:
```diff
@@ -0,0 +1,200 @@
+        >>> bike26 = Bicycle(12, 'disk', 27.5)  # инициализация экземпляра класса
+        """
+        self.init_bicycle_sp(speed)
+        self.speed = speed
```
Очень хорошо, но эта строчка тогда уж лишняя, ведь в методе после проверки вы уже создадите атрибут. Давайте PR'ы не будем плодить, и для платформы будет проще (чтобы вы не ждали возможности обновления ссылки на дл-идо, чтобы я тоже видел только одну ссылку) обновлять файлы сразу в этом PR. Вот пример
![image](https://user-images.githubusercontent.com/54230556/208024635-45953aee-fdfa-4f4e-825b-76b5995fe657.png)

frek17 replied:
Прочитал комментарий к коммиту. Ну смотрите, не знаю что там ругается PyCharm, но дважды написана одна и та же строчка кода 
`self.speed = speed`
Особого смысла в этом не вижу. Тогда уж давайте в методах не будем задавать атрибут и методы переименуем в verify

frek17 commented on main.py:
```diff
@@ -0,0 +1,200 @@
+        self.init_wheel_size(wheel_size)
+        self.wheel_size = wheel_size
+
+    def init_bicycle_sp(self, speed):
```
Хорошо бы и здесь использовать тайпхинтинги. `def init_bicycle_sp(self, speed: int) -> None:`
Добро?

frek17 commented on main.py:
```diff
@@ -0,0 +1,200 @@
+        :param speed: скорости велосипеда
+        :return: скорости велосипеда
+        """
+        if not isinstance(speed, (int, float)):
```
На вопрос-то вы не ответили)
Смотрите, давайте просто прикинем, хорошо? По сути, нам с вами нужен `int`, и черт с ним, если вдруг будет `float` (в стиле 4.0, если будет 4.7, то будет очень странно и непонятно), можно проверить тип, если он флоат, переделываем в `int`. Мои вопросы для обсуждения в первую очередь, а уже потом для исправления, не торопитесь что-то менять. Вообще говоря, можно просто строго оставлять `int`, и всё

frek17 replied:
Вопрос был скорее к тому, думали ли вы о том, что будет число с запятой

frek17 commented on main.py:
```diff
@@ -0,0 +1,200 @@
+        :return: скорости велосипеда
+        """
+        if not isinstance(speed, (int, float)):
+            raise TypeError
```
Было бы здорово ещё сообщение выводить, посмотрите пример

frek17 commented on main.py:
```diff
@@ -0,0 +1,200 @@
+
+if __name__ == "__main__":
+    doctest.testmod()
+    pass
```
Зачем `pass`, есть есть одна строчка кода? `Pass` как затычка типа `...`

frek17 commented:
Попробуйте в этих комментариях отвечать. С пендингом пока решение не вижу, попробуйте написать в общий чат или лектору, например. Одна голова хорошо, а две лучше

frek17 commented:
Ещё хочу добавить, что в принципе комментарии ко всем классом относятся, я для первого написал, но вы для второго и третьего тоже посмотрите, что там не так, хорошо?
В целом мне нравится, нэйминги хорошие, лишнего нет, подправите и будет шикарно. Так, у нас это новый PR, я вам возможность редактирования добавлю. Отпишитесь здесь потом, что ссылку обновили, чтобы точно знал, что всё хорошо

frek17 commented:
Ping

frek17 commented:
Без исправлений не могу поставить максимальный балл

MrGriv committed:
Update main.py

Относительно комментария что в методе после проверки я уже создам атрибут и поэтому его стоит удалить из __init__ PyCharm ругается, что этот атрибут остутсвует в __init__. Я хотел максимально сократить инициализацию и перенести проверку на ошибки в методы, чтобы __init__ смотрелся лаконично. 
Да, я думал, что число будет с запятой, поэтому старался выбирать такие единицы измерения, чтобы не было нужды в запятой

frek17 commented on main.py:
```diff
@@ -0,0 +1,199 @@
+        :return: скорости велосипеда
+        """
+        if not isinstance(speed, int):
+            raise TypeError('Введен неверный тип данных')
```
Хорошо бы указать, какой тип должен быть

frek17 commented on main.py:
```diff
@@ -0,0 +1,199 @@
+        if not isinstance(wheel_size, (int, float)):
+            raise TypeError('Введен неверный тип данных')
+        if wheel_size < 0:
+            raise ValueError('Число должно быть больше нуля')
```
0 тоже не может быть, включите это в условие

MrGriv commented:
Нам в лекциях просто показывали так делать. А так да, я понимаю, что два раза написана эта строчка. Для упрощения могу просто делать проверку на ошибки прямо в init

MrGriv commented:
Как должно быть?

MrGriv committed:
Update main.py
